quiche/src/lib.rs

```diff
@@ -6264,9 +6338,51 @@ impl Connection {
     pub fn is_server(&self) -> bool {
         self.is_server
     }
+    
+    /// Returns the server's preferred address which is a tuple with a
+    /// V4 socket address, a connection ID and a stateless reset token.
+    pub fn server_preferred_address_v4(
+        &self,
+    ) -> Result<Option<(SocketAddrV4, ConnectionId, u128)>> {
```

Comment:
So I'm wondering what the utility of the Ok(None) is? 

I imagine a case where a client can call the `_v4` method, sees there isn't anything so it checks the `_v6` version. 

But maybe it would be easier to just return a copy of the parameter directly to the client and let it figure this all out (i.e. it can do its own pattern matching on the things it really cares about). In other words, just return `Option<PreferredAddressParams>`.

And I'm not too sure the value of ClientOnlyOperation error. If we just returned None to a server always and documented that, would server app react to the situation any differently?